EDA blueprints

okrause opened the pull request:
This PR adds two blueprints and a README.MD file to a new examples/eda folder.

Both blueprints use Google Cloud NetApp Volumes as the shared NFS storage for the compute cluster. The eda-all-on-cloud blueprint  builds on top of #4583. Since that one is not merged, this PR looks pretty big, but it is only a folder, a readme and two blueprint files.

I open it as a draft PR for now. To make it ready for prime time, the following things need to happen:
1. PR #4583 needs to be merged to develop
2. I need to complete work on README.md. Will complete next week.
3. I need to work with Google EDA experts to have them verify (and potentially update) the cluster deployment group for EDA workloads.

okrause commented:
Hello, this PR is about the files in the examples/eda folder. It adds two new blueprints and a readme.md. All the other changes are from #4583 and not subject of this PR. I just want to get the review process started on the two blueprints and the readme file.

Kudos to @thomasnyc for providing the H4D compute part of these blueprints.

sarthakag commented on examples/README.md:
```diff
@@ -64,7 +64,7 @@ md_toc github examples/README.md | sed -e "s/\s-\s/ * /"
   * [gke-h4d](#gke-h4d-) ![core-badge]
   * [gke-g4](#gke-g4-) ![core-badge]
-  * [netapp-volumes.yaml](#netapp-volumesyaml--) ![core-badge]
+  * [netapp-volumes.yaml](#netapp-volumesyaml--) ![community-badge]
```
revert back to core?

sarthakag commented on examples/README.md:
```diff
@@ -1698,6 +1698,24 @@ To destroy all resources associated with creating the GKE cluster, run the follo
 
+### [eda-all-on-cloud] ![core-badge]
+
+Creates a basic auto-scaling Slurm cluster intended for EDA use cases. The blueprint also creates two new VPC networks, one frontend network which connects VMs, SLURM and storage and the other for fast RDMA networking between the H4D nodes, along with four [Google Cloud NetApp Volumes](https://cloud.google.com/netapp/volumes/docs/configure-and-use/volumes/overview) mounted to `/home`, `/tools`, ` ...
```
s/SLURM/slurm

sarthakag commented on examples/README.md:
```diff
@@ -1698,6 +1698,24 @@ To destroy all resources associated with creating the GKE cluster, run the follo
+
+### [eda-hybrid-cloud] ![core-badge]
+
+Creates a basic auto-scaling Slurm cluster intended for EDA use cases. The blueprint also connects to one exiting frontend network which connects VMs, SLURM and storage and creates a new RDMA network for low latency communication between the compute nodes. There is an `h4d` partition that uses compute-optimized `h4d-highmem-192-lssd` machine type.
```
s/SLURM/slurm

bytetwin commented on examples/eda/eda-all-on-cloud.yaml:
```diff
@@ -0,0 +1,236 @@
+#
+# Sets up VPC network, persistent NFS shares
+# ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+- group: setup
```
Pattern is this group is called primary or base

okrause commented on examples/eda/eda-all-on-cloud.yaml:
```diff
@@ -0,0 +1,236 @@
+    source: modules/network/vpc
+
+  # Backend RDMA network for GCE instances with RDMA capabilities
+  - id: backend-network
```
renamed

okrause commented:
> @okrause
> 
> 1. Both the blueprints and its Readme should be in community/examples. Examples folder is for core blueprints that cluster toolkit team supports and will continue to maintain. Netapp volume is gcp supported product and is in core module as it can be used as filestore, these blueprints are for specific workload type which cluster toolkit team won't support.
> 2. Please avoid adding images to the repo. Cluster toolkit is cloned from github by users as well as automated systems and repository size will impact the cloning speed.

@bytetwin Understood. Moved to community/examples/eda. Removed the pictures. Especially the hybrid use case benefit a lot from a picture, so I added a link to a blog where the interested user can find a picture of the concept. Pushed the changes.

Thank you for your input!

bytetwin commented on community/examples/eda/eda-all-on-cloud.yaml:
```diff
@@ -0,0 +1,236 @@
+    use: [eda-net, private_service_access]
+    settings:
+      pool_name: "eda-pool"
+      network_id: $(eda-net.network_id)
```
network_id is not required as the `use: [eda-net, private_service_access]` directive will take care of filling in the outputs of `eda-net` into this module

bytetwin commented on community/examples/eda/README.md:
````diff
@@ -0,0 +1,245 @@
+to confirm the deletion of each stage.
+
+```bash
+./gcluster destroy <blueprint>
````
destroy takes in the deployment folder generated by deploy command.

destroy doesn't understand blueprint, refer - https://github.com/GoogleCloudPlatform/cluster-toolkit/blob/main/examples/gke-g4/README.md#clean-up

bytetwin commented on examples/README.md:
```diff
@@ -1698,6 +1698,24 @@ To destroy all resources associated with creating the GKE cluster, run the follo
 
+### [eda-all-on-cloud] ![community-badge]
+
+Creates a basic auto-scaling Slurm cluster intended for EDA use cases. The blueprint also creates two new VPC networks, one frontend network which connects VMs, Slurm and storage and the other for fast RDMA networking between the H4D nodes, along with four [Google Cloud NetApp Volumes](https://cloud.google.com/netapp/volumes/docs/configure-and-use/volumes/overview) mounted to `/home`, `/tools`, ` ...
```
replace frontend with base or similar naming convention used in the blueprint
